aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,195 @@
+
+## Choose an ASP.NET Core web UI technology
+
+Use the following flowchart to select an ASP.NET Core web UI technology:
+
+![Decision tree for ASP.NET Core web UI technology](choose-web-ui/_static/choose_ui.png)
+
+## Choose server rendered UI or client rendered UI
+
+There are two general approaches to building modern web UI:
+
+* Apps that render UI on the server.
+* Apps that render UI on the client in the browser.
+
+There are benefits and drawbacks to consider for either approach. A hybrid approach is also possible.  For example, most of the web UI is rendered on the server, and client rendered pages are added as needed.
+
+### When to choose a server rendered UI approach
+
+A web UI app that renders on the server dynamically generates the page's HTML and CSS on the server. The page arrives at the client ready to display.
+
+Benefits:
+
+* The client requirements are minimal since the server does the work of logic and page generation:
+  * Great for low-end devices and low-bandwidth connections.
+  * Allows for a broad range of browser versions at the client.
+  * Minimal to no code to pull to the client.
+* Flexibility of access to server resources, in a protected environment. For example:
+  * Database access.
+  * Access to secrets, such as an access key used for API calls to Azure storage.
+* Uses a token-based security model, rather than using cookies.
+
+Examples of server rendered web UI apps:
+
+* A website for bank services.
+* A blog.
+* A public-facing content management system.
+
+Drawbacks:
+
+* The cost of compute and memory use are concentrated on the server, rather than each client.
+* Rich interactivity isn't instant: Rich interactive-driven changes are best generated on the client, rather than requiring a round trip to the server to generate an updated page.
+
+### When to choose a client rendered UI approach
+
+A client rendered web UI app dynamically renders UI on the client, generating and updating the page on the client.
+
+Benefits:
+
+* Takes advantage of the capabilities of the user’s device.
+* Allows for rich interactivity that is instant, without requiring a round trip to the server. UI event handling and logic runs locally on the user's device with minimal latency. Choose a client rendered application if your app's requirements include rich functionality that goes beyond what typical HTML forms offer.
+* Supports incremental updates, saving partially completed forms or documents without the user having to select a button to submit a form.
+* Can be designed to run in a disconnected mode. Updates to the client-side model that are eventually synchronized back to the server once a connection is re-established.
+
+Examples of client rendered web UI:
+
+* An interactive dashboard.
+* An app featuring drag-and-drop functionality
+* A responsive and collaborative social app.
```

Comment:
I like how this list is more focused on specific UI scenarios than broad app types.